hugr/src/ops/constant.rs

```diff
@@ -547,4 +555,99 @@ mod test {
         assert_ne!(yaml_const.const_type(), t);
     }
+
+    #[cfg(feature = "proptest")]
+    mod proptest {
+        use super::super::ExtensionValue;
+        use crate::{
+            ops::Value,
+            std_extensions::arithmetic::int_types::{ConstInt, LOG_WIDTH_MAX},
+            std_extensions::collections::ListValue,
+            types::{SumType, Type},
+        };
+        use ::proptest::prelude::*;
+        impl Arbitrary for ExtensionValue {
+            type Parameters = ();
+            type Strategy = BoxedStrategy<Self>;
+            fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+                use proptest::collection::vec;
+                let signed_strat = (..=LOG_WIDTH_MAX).prop_flat_map(|log_width| {
+                    use std::i64;
+                    let max_val = (2u64.pow(log_width as u32) / 2) as i64;
+                    let min_val = -max_val - 1;
+                    (min_val..=max_val).prop_map(move |v| {
+                        ExtensionValue::new(
+                            ConstInt::new_s(log_width, v).expect("guaranteed to be in bounds"),
+                        )
+                    })
+                });
+                let unsigned_strat = (..=LOG_WIDTH_MAX).prop_flat_map(|log_width| {
+                    (0..2u64.pow(log_width as u32)).prop_map(move |v| {
+                        ExtensionValue::new(
+                            ConstInt::new_u(log_width, v).expect("guaranteed to be in bounds"),
+                        )
+                    })
+                });
+                prop_oneof![unsigned_strat, signed_strat]
+                    .prop_recursive(
+                        3,  // No more than 3 branch levels deep
+                        32, // Target around 32 total elements
+                        3,  // Each collection is up to 3 elements long
+                        |element| {
+                            (any::<Type>(), vec(element.clone(), 0..3)).prop_map(
+                                |(typ, contents)| {
+                                    ExtensionValue::new(ListValue::new(
+                                        typ,
+                                        contents.into_iter().map(|e| Value::Extension { e }),
+                                    ))
+                                },
+                            )
+                        },
+                    )
+                    .boxed()
+            }
+        }
+
+        impl Arbitrary for Value {
+            type Parameters = ();
+            type Strategy = BoxedStrategy<Self>;
+            fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+                use ::proptest::collection::vec;
+                let leaf_strat = prop_oneof![
+                    any::<ExtensionValue>().prop_map(|e| Self::Extension { e }),
+                    prop_oneof![
+                        // TODO we need an example of each legal root, in particular FuncDe{fn,cl}
```

Comment:
I reckon in future we will have `impl Arbitrary for Hugr` - nothing says it has to be able to generate *any* Hugr, but we could cover a good cross-section - but very happy not to do that in this PR